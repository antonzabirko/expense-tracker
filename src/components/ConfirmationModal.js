import React from 'react';
import Modal from 'react-modal';

const ConfirmationModal = (props) => (
    <Modal
        className="modal"
        isOpen={props.modalState}
        onRequestClose={props.onModalCancel}
    >
        <div className="modal__border-container">
            <h1 className="modal__title">Remove Expense</h1>
            <p className="modal__body">Proceeding will delete and un-archive this expense.</p>
            <div className="button__container">
                <button className="button button-secondary" onClick={props.onModalConfirm}>Proceed</button>
                <button className="button" onClick={props.onModalCancel}>Cancel</button>
            </div>
        </div>
    </Modal>
);

export default ConfirmationModal;