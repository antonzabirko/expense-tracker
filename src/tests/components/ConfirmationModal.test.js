import React from 'react';
import { shallow } from 'enzyme';
import ConfirmationModal from '../../components/ConfirmationModal';

test('Should render ConfirmationModal', () => {
    const wrapper = shallow(<ConfirmationModal modalState={true}/>);

    expect(wrapper).toMatchSnapshot();
});