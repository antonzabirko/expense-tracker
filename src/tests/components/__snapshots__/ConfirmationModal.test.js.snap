// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ConfirmationModal 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={0}
  isOpen={true}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="modal__border-container"
  >
    <h1
      className="modal__title"
    >
      Remove Expense
    </h1>
    <p
      className="modal__body"
    >
      Proceeding will delete and un-archive this expense.
    </p>
    <div
      className="button__container"
    >
      <button
        className="button button-secondary"
      >
        Proceed
      </button>
      <button
        className="button"
      >
        Cancel
      </button>
    </div>
  </div>
</Modal>
`;
